Makefile

```diff
@@ -84,25 +76,40 @@ fmt:
 	go fmt $(TEST_PKGS)
 
 container: hyperfed
-	cp -f $(HYPERFED_TARGET) images/federation-v2/hyperfed
+	cp -f $(HYPERFED_TARGET)-linux images/federation-v2/hyperfed
 	$(DOCKER) build images/federation-v2 \
 		-t $(REGISTRY)/$(TARGET):$(GIT_VERSION)
 	rm -f images/federation-v2/hyperfed
 
 $(BIN_DIR):
 	mkdir $(BIN_DIR)
 
-hyperfed: $(BIN_DIR)
-	$(DOCKER_BUILD) '$(BUILD_HYPERFED)'
+$(HYPERFED_TARGET)-linux: $(BIN_DIR)
+	$(DOCKER_BUILD) '$(TARGET_ENV_LINUX) $(GO_BUILDCMD) -o $(HYPERFED_TARGET)-linux cmd/hyperfed/main.go'
+
+$(HYPERFED_TARGET)-darwin: $(BIN_DIR)
```

Comment:
(suggestion) Maybe we could have reduced duplicate lines if the Makefile target is something like `$(HYPERFED_TARGET)-$(HOST_OS): `

Reply:
i made some changes along the line.
https://github.com/kubernetes-sigs/federation-v2/pull/684/commits/5046bf44b63abdb21dce502c43f4797be1a9cd86